c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -155,6 +155,19 @@ impl<'tcx> intravisit::Visitor<'tcx> for ConvertVisitor<'tcx> {
                     Rewrite::Ref(Box::new(self.get_subexpr(ex, 0)), mutbl_from_bool(*mutbl))
                 }
 
+                mir_op::RewriteKind::CastRefToRaw { mutbl } => {
+                    let rw_pl = Rewrite::Deref(Box::new(hir_rw));
+                    Rewrite::AddrOf(Box::new(rw_pl), mutbl_from_bool(*mutbl))
+                }
+                mir_op::RewriteKind::CastRawToRaw { to_mutbl } => {
+                    let ty = if *to_mutbl { "*mut _" } else { "*const _" };
```

Comment:
If `CastRawToRaw` is meant only for `const` to `mut` and vice versa casts (as opposed to transmuting casts between different types), I'm worried the `_` could be inferred in some contexts to be another type (I'm not sure exactly how the inference rules work).  Given we don't have the full type here to use, what about using [`.cast_const`](https://doc.rust-lang.org/std/primitive.pointer.html#method.cast_const) and [`.cast_mut`](https://doc.rust-lang.org/std/primitive.pointer.html#method.cast_mut).  These methods were added for this reason: to avoid accidentally changing the type.

Reply:
Good idea - I changed it to use these methods.